### PR TITLE
feat: Add DropRules Support

### PR DESCRIPTION
**Pull request by paul91**
Me attempting to implement #659 by learning the New Relic development tool kit. I have very little Go experience so any feedback is greatly appreciated. This is work to hopefully expedite the addition of https://github.com/newrelic/terraform-provider-newrelic/issues/1078.

This is a WIP as I could use some direction on how to handle the seemingly not fully implemented GraphQL mutations/queries around `nrqlDropRules`, these would answer the TODOs:

- There is no GET query, only LIST. Should I just create my own GET that runs LIST and then pulls the desired rule out of the list? Or is there a better way to do this?
- There is no UPDATE mutation, should this just be skipped?

Still pending:

- DELETE
- A `example_basic_test.go`

### CODE REVIEW

**Comment by jthurman42** [comment]
This is awesome! Slight glare with #665 time-wise, which I'm happy to close in place of this.  There were some tutone changes that need to be made to streamline the process as well.  Once those are in place, you should be good to go.

File: .tutone.yml [hunk 1]
```diff
@@ -544,6 +544,36 @@ packages:
         field_type_override: nrtime.DateTime
         skip_type_create: true
 
+  - name: droprules
+    path: pkg/droprules
+    import_path: github.com/newrelic/newrelic-client-go/pkg/droprules
+    generators:
+      - typegen
```
Comment (jthurman42) [inline .tutone.yml:551]: Check out [this PR](https://github.com/newrelic/newrelic-client-go/pull/665/files#diff-c8fb5f06f26ca2ea1554da789ba1914c18e334991add8cbd66482c4568907a17R598) for auto-generating the queries / mutations as well (we're trying to do that as much as possible)

File: .tutone.yml [hunk 2]
```diff
@@ -544,6 +544,36 @@ packages:
...
+      - name: NrqlDropRulesListDropRulesResult
+      - name: NrqlDropRulesDropRule
+      - name: ID
+        field_type_override: string
+        skip_type_create: true
+
+      #
+      # Types that we should auto-detect are in another package someday
+      #
+      - name: DateTime
```
Comment (jthurman42) [inline .tutone.yml:574]: 👍

**Comment by jthurman42** [comment]
This is awesome! Slight glare with #665 time-wise, which I'm happy to close in place of this.  There were some newrelic/tutone#150 changes that need to be made to streamline the process for generating the API code.  Once those are in place, you should be able to auto-generate most of the API level stuff, including DELETE

**Comment by paul91** [comment]
Amazing, will check that out tomorrow, thanks!

**Comment by paul91** [comment]
@jthurman42 I went ahead and squashed our changes together for fun. I am happy with either effort to go forward, just let me know what makes the most sense.

Only issue I am seeing is that `NRQLDropRuleCreate` actually throws an error as ID comes back as a string instead of an int so it can't marshal as expected:

```
    nrqldroprules_integration_test.go:45: 
                Error Trace:    nrqldroprules_integration_test.go:45
                Error:          Received unexpected error:
                                json: cannot unmarshal string into Go struct field NRQLDropRulesDropRule.data.NRQLDropRulesCreate.successes.id of type int
                Test:           TestIntegrationDropRules
```

Again, feel free to nit or replace my changes as needed. Thanks!

**Comment by zlesnr** [comment]
I suspect in the tutone config, you just need something to override the default type.  Something like https://github.com/newrelic/newrelic-client-go/blob/master/.tutone.yml#L67 might do it.

File: .tutone.yml [hunk 3]
```diff
@@ -590,6 +590,57 @@ packages:
...
+      - name: NrqlDropRulesDeleteDropRuleFailure
+      - name: NrqlDropRulesDeleteDropRuleResult
+      - name: NrqlDropRulesDeleteDropRuleSubmission
+      - name: NrqlDropRulesDropRule
+      - name: NrqlDropRulesDropRule
+      - name: NrqlDropRulesError
+      - name: NrqlDropRulesErrorReason
+      - name: NrqlDropRulesListDropRulesResult
+      # All of the IDs are created as ints, yet this unused type is being created.
+      - name: ID
```
Comment (jthurman42) [inline .tutone.yml:628]: This likely needs `field_type_override: string` to fix the unmarshall issue.

**Comment by jthurman42** [comment]
@paul91 If you're read, please rebase and we'll get this merged in.

**Comment by paul91** [comment]
Done. Not sure what I need to do with this `synk` test though.

**Comment by jthurman42** [comment]
@paul91 No worries on Snyk, security scan won't run on a remote branch for security reasons. Happens on all contributor PRs which is why it is not a required check.

**Review by jthurman42 — APPROVED** [review]
:shipit:

**Comment by jthurman42** [comment]
Thank you so much for the contribution! If you're feeling up to it, there is an open issue to add this into the New Relic Terraform Provider as well newrelic/terraform-provider-newrelic#1078 :)

**Comment by paul91** [comment]
Thanks for the review! My coworker originated that request, so will take a shot at it ASAP.

**Pull request merged**